doc/contributor/code-patterns.md

```diff
@@ -0,0 +1,52 @@
+# Code Patterns
+
+## Where to allocate helper nodes (a node used in another node)
+
+* If the node does not use the DSL, you should allocate the helper node [lazily](#the-lazy-pattern).
```

Comment:
`If the node`  is the "node" the child helper node or the helper's parent?

`use the DSL`  does this mean is it using `@Specialization` ?

Reply:
I mean the parent/current node here.

> use the DSL does this mean it is using @Specialization ?

Yes.